### PR TITLE
Clean up run_analysis method for nf_analysis

### PR DESCRIPTION
**Pull request by fevac**
### Changed

- Split run_analysis method

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [x] `cg workflow rnafusion run -d CASE`
- [x] `cg workflow rnafusion run -d --use-nextflow CASE`
- [x] `cg workflow rnafusion run -d --from-start CASE`

### Expected test outcome

- [x] run command shows correctly

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/meta/workflow/nf_analysis.py [hunk 1]
```diff
@@ -205,6 +205,66 @@ def write_trailblazer_config(self, case_id: str, tower_id: str) -> None:
...
+        self, case_id: str, command_args: CommandArgs, dry_run: bool
+    ) -> None:
+        """Run analysis with given options using Nextflow."""
+        self.process = Process(
+            binary=self.nextflow_binary_path,
+            environment=self.conda_env,
+            conda_binary=self.conda_binary,
+            launch_directory=self.get_case_path(case_id=case_id),
+        )
+        LOG.info("Pipeline will be executed using nextflow")
```
Comment (henrikstranneheim) [inline cg/meta/workflow/nf_analysis.py:218]:
```suggestion
        LOG.info("Pipeline will be executed using Nextflow")
```

File: cg/meta/workflow/nf_analysis.py [hunk 2]
```diff
@@ -205,6 +205,66 @@ def write_trailblazer_config(self, case_id: str, tower_id: str) -> None:
...
+        parameters: list[str] = NextflowHandler.get_nextflow_run_parameters(
+            case_id=case_id,
+            pipeline_path=self.nfcore_pipeline_path,
+            root_dir=self.root_dir,
+            command_args=command_args.dict(),
+        )
+        self.process.export_variables(
+            export=NextflowHandler.get_variables_to_export(),
+        )
+        command = self.process.get_command(parameters=parameters)
```
Comment (henrikstranneheim) [inline cg/meta/workflow/nf_analysis.py:228]: Add type hint

File: cg/meta/workflow/nf_analysis.py [hunk 3]
```diff
@@ -205,6 +205,66 @@ def write_trailblazer_config(self, case_id: str, tower_id: str) -> None:
...
+        sbatch_number: int = NextflowHandler.execute_head_job(
+            case_id=case_id,
+            case_directory=self.get_case_path(case_id=case_id),
+            slurm_account=self.account,
+            email=self.email,
+            qos=self.get_slurm_qos_for_case(case_id=case_id),
+            commands=command,
+            dry_run=dry_run,
+        )
+        LOG.info(f"Nextflow head job running as job {sbatch_number}")
```
Comment (henrikstranneheim) [inline cg/meta/workflow/nf_analysis.py:239]:
```suggestion
        LOG.info(f"Nextflow head job running as job: {sbatch_number}")
```

File: cg/meta/workflow/nf_analysis.py [hunk 4]
```diff
@@ -205,6 +205,66 @@ def write_trailblazer_config(self, case_id: str, tower_id: str) -> None:
...
+
+    def _run_analysis_with_tower(
+        self, case_id: str, command_args: CommandArgs, dry_run: bool
+    ) -> None:
+        """Run analysis with given options using NF-Tower."""
+        LOG.info("Pipeline will be executed using tower")
```
Comment (henrikstranneheim) [inline cg/meta/workflow/nf_analysis.py:245]:
```suggestion
        LOG.info("Pipeline will be executed using Tower")
```

**Review by seallard — APPROVED** [review]

File: cg/meta/workflow/nf_analysis.py [hunk 5]
```diff
@@ -205,6 +205,66 @@ def write_trailblazer_config(self, case_id: str, tower_id: str) -> None:
...
+        if command_args.resume:
+            from_tower_id: int = command_args.id or NfTowerHandler.get_last_tower_id(
+                case_id=case_id,
+                trailblazer_config=self.get_trailblazer_config_path(case_id=case_id),
+            )
+            LOG.info(f"Pipeline will be resumed from run {from_tower_id}.")
```
Comment (henrikstranneheim) [inline cg/meta/workflow/nf_analysis.py:251]:
```suggestion
            LOG.info(f"Pipeline will be resumed from run with Tower id: {from_tower_id}.")
```

**Review by henrikstranneheim — APPROVED** [review]
Nice 👍

File: cg/meta/workflow/nf_analysis.py (hunk 2 above)
Comment (fevac) [reply cg/meta/workflow/nf_analysis.py:228]: added

**Comment by fevac** [comment]
tests are passing

**Pull request merged**